prov/tcp/src/tcpx_ep.c

```diff
@@ -185,6 +200,42 @@ static int tcpx_ep_shutdown(struct fid_ep *ep, uint64_t flags)
 	return ret;
 }
 
+static int tcpx_pep_bind_to_port_range(struct tcpx_pep *pep)
+{
+	int ret, i, rand_port_number;
+
+	srand(getpid());
+	rand_port_number = rand() % (pep->port_range.high + 1 - pep->port_range.low) +
+			   pep->port_range.low;
+
+	for (i = pep->port_range.low; i <= pep->port_range.high;
+	     i++, rand_port_number++) {
+		if (rand_port_number > pep->port_range.high) {
+			rand_port_number = pep->port_range.low;
+		}
+		ofi_addr_set_port(pep->info->src_addr, rand_port_number);
+		ret = bind(pep->sock, pep->info->src_addr,
+		      (socklen_t) pep->info->src_addrlen);
+		if (ret) {
+			if (errno == EADDRINUSE) {
+				continue;
+			} else {
+				FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL,
+					"failed to bind listener: %s\n",
+					strerror(ofi_sockerr()));
+				goto err;
```

Comment:
Use
```
return -errno
```
 to be consistent with other functions, where negative errno is returned in case of failures.